packages/react-instantsearch-hooks-web/src/widgets/SearchBox.tsx

```diff
@@ -12,14 +12,30 @@ type UiProps = Pick<
   | 'isSearchStalled'
   | 'onChange'
   | 'onReset'
+  | 'onSubmit'
   | 'value'
   | 'translations'
 >;
 
-export type SearchBoxProps = Omit<SearchBoxUiComponentProps, keyof UiProps> &
-  UseSearchBoxProps;
+export type SearchBoxProps = Omit<
+  SearchBoxUiComponentProps,
+  Exclude<keyof UiProps, 'onSubmit'>
+> &
+  UseSearchBoxProps & {
+    /**
+     * If false, triggers the search only on submit.
+     * @type {boolean}
+     * @default true
+     * @optional
```

Comment:
We rely on TypeScript for type and optional indications. Otherwise we get duplications and it's harder to maintain.

```suggestion
     * Whether to trigger the search only on submit.
     * @default true
```

Reply:
Makes sense, thanks 😄